docs/devel/test-nuclio-locally.md

```diff
@@ -0,0 +1,83 @@
+# Running Nuclio on your computer with local changes
+
+This guide will guide you through the process of setting up Nuclio locally for testing your changes and provide faster development cycles.
+
+#### In This Document
+
+- [Prerequisites](#prerequisites)
+
+- [Running Nuclio](#running-nuclio)
+
+<a id="prerequisites"></a>
+## Prerequisites
+
+Ensure that your setup includes the following prerequisite components:
+
+- Linux or OSX
+- Git
+- Docker version 19.03+
+- The Go toolchain (CI tests with 1.17, best use that)
+- Kubernetes version 1.20+ (mostly for testing) - `minikube` recommended; (you can follow
+  the [Minikube getting-started guide](/docs/setup/minikube/getting-started-minikube.md))
+- Node version 10.x
+- Goland IDE
+
+<a id="running-nuclio"></a>
+## Running Nuclio
+
+1. Install Nuclio CRDs. You can install them by running `test/k8s/ci_assets/install_nuclio_crds.sh`.
+
+
+2. Create a configs folder outside of the Nuclio repo. Inside it, create `configs/platform_config.yaml` with the following body:
+```yaml
+logger:
+  sinks:
+    myStdoutLoggerSink:
+      kind: stdout
+  system:
+    - level: debug
+      sink: myStdoutLoggerSink
+  functions:
+    - level: debug
+      sink: myStdoutLoggerSink
+
+kube:
+  kubeConfigPath: <your-home-dir>/.kube/config
+
+containerBuilderConfiguration:
+  DefaultOnbuildRegistryURL: "localhost:5000"
+```
+
+3. Run local registry:
+```sh
+docker run --rm -d -p 5000:5000 registry:2
+```
+4. Run Dashboard:
+   1. Open `cmd/dashboard/main.go`
+   2. In Goland - create a run configuration for the `main` function with the following program arguments:
+   ```sh
+   --platform kube --platform-config path/to/configs/platform_config.yaml --namespace default --registry localhost:5000 --run-registry localhost:5000
+   ```
```

Comment:
we can add this run configuration (to even multiple, one for local one for kube, etc) to the source code (when you edit, look at the top right - 

<img width="160" alt="Screen Shot 2022-01-06 at 17 18 10" src="https://user-images.githubusercontent.com/6547459/148405510-ce27bce1-f826-49a5-b83e-d6bec0ccd8b3.png">

make sure the enviroments/arguments do not have sensitive information.